controls/aws-foundations-cis-2.1.4.rb

```diff
@@ -0,0 +1,216 @@
+https://console.aws.amazon.com/s3/
+2. Choose `Block Public Access (account
+settings)`
+3. Choose `Edit` to change the block public access settings for all the buckets
+in your AWS account
+4. Choose the settings you want to change, and then choose `Save`. For
+details about each setting, pause on the `i` icons.
+5. When you're asked for confirmation,
+enter `confirm`. Then Click `Confirm` to save your changes.
+
+**From Command
+Line:**
+
+To set Block Public access settings for this account, run the following
+command:
+```
+aws s3control
+put-public-access-block
+--public-access-block-configuration
+BlockPublicAcls=true, IgnorePublicAcls=true, BlockPublicPolicy=true,
+RestrictPublicBuckets=true
+--account-id <value>
+``` "
+  desc 'impact',
+       "When you apply Block Public Access settings to an account, the settings apply to all AWS
+Regions globally. The settings might not take effect in all Regions immediately or
+simultaneously, but they eventually propagate to all Regions. "
+  impact 0.5
+  ref 'https://docs.aws.amazon.com/AmazonS3/latest/user-guide/block-public-access-account.html'
+  tag nist: ['AC-3']
+  tag severity: 'medium '
+  tag cis_controls: [{ '8' => ['3.3'] }]
+
+  exempt_buckets = input('exempt_buckets')
+  s3_buckets = aws_s3_buckets.bucket_names
+  failing_buckets = []
+
+  describe 'No Tests Defined Yet' do
+    skip 'No Tests have been written for this control yet'
+  end
+
+  # # EITHER the AWS account itself should be blocking public access, OR each individual bucket should do so
+  # describe.one do
+  #   describe aws_public_access_settings do
+  #     its('IgnorePublicAcls') { should eq "true" }
+  #     its('BlockPublicPolicy') { should eq "true" }
+  #     its('BlockPublicAcls') { should eq "true" }
+  #   end
+  #   only_if('This control is Non Applicable since no unexempt S3 buckets were found.', impact: 0.0) { !s3_buckets.empty? or !(exempt_buckets - s3_buckets).empty? }
+
+  #   if input('single_bucket').present?
+  #     failing_buckets << input('single_bucket').to_s unless aws_s3_bucket(bucket_name: input('single_bucket')) # TODO: find the right filter
+  #     describe "The #{input('single_bucket')}" do
+  #       it 'should block public access' do
+  #         expect(failing_buckets).to be_empty, "Failing buckets:\t#{failing_buckets}"
+  #       end
+  #     end
+  #   else
+  #     failing_buckets = s3_buckets.select { |bucket|
+  #       next if exempt_buckets.include?(bucket)
+  #       !aws_s3_bucket(bucket_name: bucket) #TODO: find the right filter
+  #     }
+  #     describe 'S3 buckets' do
+  #       it 'should all block public access' do
+  #         failure_messsage = "Failing buckets:\n#{failing_buckets.join(", \n")}"
+  #         failure_messsage += "\nExempt buckets:\n\n#{exempt_buckets.join(", \n")}" if exempt_buckets.present?
+  #         expect(failing_buckets).to be_empty, failure_messsage
+  #       end
+  #     end
+  #   end
+  # end
+end
```

Comment:
Opened a pull request on train-aws to update all gems to get arround the Parse URI errors https://github.com/inspec/train-aws/pull/519